packages/expect/src/matchers.ts

```diff
@@ -62,32 +70,55 @@ const matchers: MatchersObject = {
       ? () =>
           matcherHint(matcherName, undefined, undefined, options) +
           '\n\n' +
-          `Expected: ${printExpected(expected)}\n` +
-          `Received: ${printReceived(received)}`
+          `Expected: not ${printExpected(expected)}`
       : () => {
           const receivedType = getType(received);
           const expectedType = getType(expected);
-          const suggestToEqual =
-            receivedType === expectedType &&
-            (receivedType === 'object' || expectedType === 'array') &&
-            equals(received, expected, [iterableEquality]);
-          const oneline = isOneline(expected, received);
-          const diffString = diff(expected, received, {expand: this.expand});
+
+          const isShallow =
+            receivedType !== expectedType ||
+            (isPrimitive(expected) &&
+              (expectedType !== 'string' || isOneline(expected, received))) ||
+            expectedType === 'function' ||
+            expectedType === 'regexp' ||
+            (received instanceof Error && expected instanceof Error);
+
+          const deepEqualityName =
```

Comment:
Small nit: I think null instead of the empty string (adapt `===` in L97 and use ternary in return) would improve readability, esp. given that `difference` is also used that way.

Reply:
Yes, it does improve readability!